core/src/main/groovy/com/novoda/gradle/release/AndroidArtifacts.groovy

```diff
@@ -1,42 +1,48 @@
 package com.novoda.gradle.release
 
 import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.component.SoftwareComponent
 import org.gradle.api.tasks.bundling.Jar
 import org.gradle.api.tasks.javadoc.Javadoc
 
 class AndroidArtifacts implements Artifacts {
 
     def variant
 
+    // TODO: Declare that variant is
+    // https://google.github.io/android-gradle-dsl/current/com.android.build.gradle.LibraryExtension.html#com.android.build.gradle.LibraryExtension:libraryVariants
     AndroidArtifacts(variant) {
         this.variant = variant
     }
 
+    @Override
     def all(String publicationName, Project project) {
         [sourcesJar(project), javadocJar(project), mainJar(project)]
     }
 
     def sourcesJar(Project project) {
-        project.task(variant.name + 'AndroidSourcesJar', type: Jar) {
-            classifier = 'sources'
+        project.tasks.create("${variant.name}AndroidSourcesJar", Jar) {
+            it.classifier = 'sources'
             variant.sourceSets.each {
                 from it.java.srcDirs
             }
         }
     }
 
     def javadocJar(Project project) {
-        def androidJavadocs = project.task(variant.name + 'AndroidJavadocs', type: Javadoc) {
+        Task androidJavadocs = project.tasks.create("${variant.name}AndroidJavadocs", Javadoc) {
             variant.sourceSets.each {
                 delegate.source it.java.srcDirs
             }
-            classpath += project.files(project.android.getBootClasspath().join(File.pathSeparator))
-            classpath += variant.javaCompile.classpath
-            classpath += variant.javaCompile.outputs.files
+            it.classpath += project.files(project.android.getBootClasspath().join(File.pathSeparator))
```

Comment:
Same consideration here about naming the input instead of relying on the implicit `it` (and all the other places below). Is this done to help the IDE resolving the type?